### ISSUE TITLE
Discord stops accepting mouse input after its window is resized

### ISSUE THREAD

**Issue by estokes** [pop-os/cosmic-epoch#665]
On PopOs 22.04 with the latest cosmic debs as of 2024-08-13 using the scroll wheel in discord causes Xwayland to crash (all X11 windows become just title bars with no content and cannot be closed). Cosmic does not restart Xwayland after that and the only way to restore X11 functionality is to log out and back in.

discord output
[181:0813/124034.181059:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 1 times!
[181:0813/124040.253123:ERROR:gl_surface_presentation_helper.cc(260)] GetVSyncParametersIfAvailable() failed for 2 times!
[5:0813/124049.287721:FATAL:electron_browser_main_parts.cc(502)] Failed to shutdown.
[181:0813/124049.287622:ERROR:connection.cc(61)] X connection error received.
Connection lost to X server ':1'

journal
Aug 13 12:40:49 mazikeen cosmic-comp[253704]: Event thread exiting due to connection error Connection reset by peer (os error 104)
Aug 13 12:40:49 mazikeen systemd[2165]: app-flatpak-com.discordapp.Discord-255813.scope: Consumed 14.704s CPU time.
Aug 13 12:40:50 mazikeen systemd[1]: Started crash report submission.
Aug 13 12:40:50 mazikeen whoopsie[256143]: [12:40:50] Using lock path: /var/lock/whoopsie/lock
Aug 13 12:40:50 mazikeen systemd[1]: whoopsie.service: Deactivated successfully.
Aug 13 12:40:51 mazikeen systemd[1]: Started crash report submission.
Aug 13 12:40:51 mazikeen whoopsie[256147]: [12:40:51] Using lock path: /var/lock/whoopsie/lock
Aug 13 12:40:51 mazikeen systemd[1]: whoopsie.service: Deactivated successfully.
Aug 13 12:41:31 mazikeen cosmic-comp[253704]: Failed to update Xwm stacking order.
Aug 13 12:41:31 mazikeen cosmic-comp[253704]: Failed to update Xwm stacking order.

cosmic-comp then outputs Failed to update Xwm stacking order several times per second until it is restarted.

**Comment by mmstick** [comment]
Do you have `popdev:master` added to the system?

**Comment by estokes** [comment]
I did not, I do now (sorry I didn't know about it). So far I can't reproduce it with the popdev:master packages. I will reopen if I manage to.

**Comment by estokes** [comment]
eeeeh, I spoke too soon. Now, it doesn't crash Xwayland, but switching to other windows and then back to discord causes discord to stop accepting mouse clicks. Typing still works, and the gui is still updating, but the mouse does nothing.

**Comment by estokes** [comment]
I can reproduce it reliably now. Steps,

1. start discord (flatpak run com.discordapp.Discord) and wait for it to display content
2. note that discord accepts mouse clicks and scroll events
3. resize the window
4. note that discord no longer accepts mouse clicks or scroll events

**Comment by ChaoHunter** [comment]
I encountered the same issue after following these steps. 

1. tart discord (flatpak run com.discordapp.Discord) and wait for it to display content
2. note that discord accepts mouse clicks and scroll events
3. resize the window
4. discord no longer accepts mouse clicks or scroll events

A workaround I found was to resize the application, switch to full screen, and then exit full screen.

**Comment by KwisatzJim** [comment]
this behavior happens with Brave Browser as well.  the workaround works on it too.  I've added popdev:master to my install

**Comment by edrodgers731** [comment]
The specified workaround did not work for me.  After updating and restarting, I have the same very consistent behavior of resizing a discord (or brave) window, and effectively locking the window until I move it. 

I see no errors when resizing, but when moving a window, this appears:

Oct 05 11:53:09 thresher cosmic-comp[2029]: [GL] GL_INVALID_VALUE error generated. Size and/or offset out of range.


❯ sudo apt-manage
Configured Sources:
pop-os-apps - Pop_OS Applications
pop-os-release - Pop_OS Release Sources
system - Pop_OS System Sources
popdev-master - Pop Development Branch master
google-cloud-sdk - google-cloud-sdk
hashicorp - hashicorp

**Comment by boswelja** [comment]
I've noticed this with Teams for Linux as well as Discord - resizing means I cannot interact with the window, but if I simply move the window after resize I can interact again

Please let me know if there's any information I can provide to help debug this problem

**Comment by boswelja** [comment]
It seems like this is no longer an issue! There are some other issues with Electron apps, but separate issues for those seem reasonable.

**Comment by KwisatzJim** [comment]
> It seems like this is no longer an issue! There are some other issues with Electron apps, but separate issues for those seem reasonable.

Yes, I just checked with Brave Browser, as that was the app I was seeing the problem with, and the problem was no longer happening.  Works great 👍🏻👍🏻👍🏻👍🏻
